tests/test_cli_confirm_exec.py

```
@@ -0,0 +1,78 @@
+def test_decline_execution(monkeypatch):
+    from gui_agents.s3 import cli_app
+
+    agent = OneStepAgent("print('no')")
+
+    called = []
+
+    def fake_execute(code_str):
+        called.append(code_str)
+
+    monkeypatch.setattr(cli_app, "execute_code", fake_execute)
+    monkeypatch.setattr(cli_app, "show_permission_dialog", lambda code, desc: False)
```

Ruff (0.14.10)

Unused lambda argument: `code`

(ARG005)